main.js

```diff
@@ -1,170 +1,161 @@
 }
 
-function escribirMensaje(mensaje, tipo){
-    $mensaje.textContent= mensaje;
-    if(tipo==="maquina"){
+function escribirMensaje(mensaje, tipo) {
+    $mensaje.textContent = mensaje;
+    if (tipo === "maquina") {
         $mensaje.classList.remove("bg-primary");
         $mensaje.classList.remove("bg-warning");
         $mensaje.classList.add("bg-success");
     }
-    else if(tipo==="jugador"){
+    else if (tipo === "jugador") {
         $mensaje.classList.remove("bg-success");
         $mensaje.classList.add("bg-warning");
     }
-    else if(tipo==="perdiste"){
+    else if (tipo === "perdiste") {
         $mensaje.classList.remove("bg-warning");
         $mensaje.classList.add("bg-primary");
     }
 }
 
-function imprimirRonda(){
-    let $numeroRonda=document.querySelector(".numero-ronda");
-    if(ronda<10){
-        $numeroRonda.textContent=`0${ronda}`;
-    }
-    else if(ronda>=10){
-        $numeroRonda.textContent= ronda;
-    }
+function imprimirRonda() {
+    const $numeroRonda = document.querySelector(".numero-ronda");
+    const imprimirNRonda = ronda.toString().padStart(2, "0");
+    $numeroRonda.textContent = imprimirNRonda;
+
 }
 
-function resaltar (element, index){
-    setTimeout(function(){
-    element.classList.add("activo");
-    }, (index+1)*1000);
+function resaltar(element, index) {
+    setTimeout(function () {
+        element.classList.add("activo");
+    }, (index + 1) * 1000);
     setTimeout(function () {
         element.classList.remove("activo");
-      }, (index+1.5)*1000);
+    }, (index + 1.5) * 1000);
 }
 
-function tomarJugadaUsuario(element){
-    $jugadaUsuario.push(element);
-    console.log($jugadaUsuario);
-    return $jugadaUsuario;
+function tomarJugadaUsuario(element) {
+    jugadaDeUsuario.push(element);
+    return jugadaDeUsuario;
 }
 
-function jugadaMaquina(){
+function jugadaMaquina() {
     escribirMensaje("turno maquina", "maquina");
     imprimirRonda();
-    let valor;
-    function azar(){valor = Math.floor(Math.random()*10); return valor;}
-    azar();
-    while(valor >3){
-        azar();
-    }
-    if(valor<=3){
-        $jugadaMaquina.push($cuadro[valor]);
-        $jugadaMaquina.forEach(resaltar);
-        console.log($jugadaMaquina);
-        setTimeout(turnoUsuario, ($jugadaMaquina.length+1) *1000);
-        for (let i = 0; i < $jugadaMaquina.length; i++) {
-            const element = $jugadaMaquina[i];
-            resaltar(element, i);
-        }
-        
-        return $jugadaMaquina;
+    let valor = generarNumRandom();
+
+    jugadaDeMaquina.push($cuadros[valor]);
+    jugadaDeMaquina.forEach(resaltar);
+    setTimeout(turnoUsuario, (jugadaDeMaquina.length + 1) * 1000);
+    for (let i = 0; i < jugadaDeMaquina.length; i++) {
+        const element = jugadaDeMaquina[i];
+        resaltar(element, i);
     }
 
-    
+    return jugadaDeMaquina;
+}
+
+function generarNumRandom() {
+    const valor = Math.floor(Math.random() * 4);
+    return valor;
 }
```

Comment:
No abrevies los nombres de funciones.

Por otro lado, ese "4" es un número mágico. Qué es ese 4? El número de colores disponibles (el número de cuadros disponibles)

Por otro lado, vos no querés generar un número random... o mejor dicho, esto genera un índice el cual después vas a utilizar para obtener un cuadro random, que es realmente tu objetivo.

Entonces en vez de eso, te propongo que a esta función la llames `obtenerCuadroAleatorio` y que tome un listado de cuadros.

```suggestion
function obtenerCuadroAleatorio($cuadros) {
    const indiceAleatorio = Math.floor(Math.random() * $cuadros.length);
    return $cuadros[indiceAleatorio];
}
```

Notá que ahora todo tiene más sentido, estás eligiendo `un cuadro aleatorio` en base a la `cantidad de cuadros disponibles` en vez de elegir un número random que iba de 1 a 4, vaya uno a saber por qué 😄 

Luego en el turno de la máquina quites estas líneas:

```
    let valor = generarNumRandom();
    jugadaDeMaquina.push($cuadros[valor]);
```

Y las reemplaces por:
```
    jugadaDeMaquina.push(obtenerCuadroAleatorio($cuadros));
```